### ISSUE TITLE
Rychlost nahrávání plného textu (PDF)

### ISSUE THREAD

**Issue by tumovam** [proarc/proarc-client#650]
V novém klientovi se potýkáme s nižší rychlostí nahrávání PDF např. k eČlánkům. 
Na schůzce 18.11. domluveno, že by mohlo pomoci rozdělit procesy na:
1. nahrávání plného textu = PDF přes fci "Načíst nový digitální obsah"
2. generování thumbnailů "Ikony" první stránky pro Krameria, kdy proces generování náhledu pojede na pozadí

Děkujeme

**Comment by SykoraLukas** [comment]
Ano, domluveno, že tyto procesy rozdělit, ale ne, že se to bude dělat ještě letos. Proto dávám label `Návrh k rozšíření`.

**Comment by SykoraLukas** [comment]
Volání nové funkce z bodu 2:

```
### GENERATE THUMBNAIL
POST {{proarcUrl}}/rest/v2/object/thumb
Accept: application/json
Content-Type: application/x-www-form-urlencoded; charset=UTF-8

pid={{uuid}}
```

Nutné přidat do lokalizačních souborů nový klíč: `externalProfile.thumbnail` s hodnotou `Externí - generování Thumbnail`

@tumovam jak se má jmenovat funkce na generování Thumbnailů? Navrhuji `Generovat Thumbnail` a dát ji vedle funkce `Generovat PDF-A`.

**Comment by tumovam** [comment]
@SykoraLukas ještě mi není jasná jedna věc. Znamená to tedy, že před exportem budeme muset vždy kliknout na "Generovat Thumbnail", aby se vygeneroval náhled pro Krameria? Nebo když vyexportuji číslo/ročník, tak se náhledy automaticky dogenerují samy? Díky

**Comment by tumovam** [comment]
Zkouším si to představit v praxi. A nejsem si jistá, jestli bude stačit ručně vygenerovat třeba jen přední obálku čísla, nebo náhled pro každý článek. V tom druhém případě by se asi zpracovatelé uklikali a obávám se, že i občas chybovali a něco nevygenerovali. Nevím, jak si s tím poradí Kramerius, co by ukazoval v případě, kdy nedostane náhled.

**Comment by SykoraLukas** [comment]
Když nad tím přemýšlím tak mi to přijde také nepraktické. Co to udělat jinak? Tu funkci tam nechat, ale zároveň po nahrání PDF uložit to PDF do objektu, jako je již  nyní, ale místo automatického generování thumbnailu (což je aktuální stav) a místo nutnosti spustit funkci (což je stav požadovaný v zadání issue) by se jen založil nový proces na generování thumbnailu a uživateli se vrátila hláška, že pdf bylo nahráno a proces zapsán.

Generovalo by se to pořád automaticky, takže by uživatel nemusel na nic klikat, ale zároveň by byla jistota, že to bude všude a zároveň by to nemělo zdržovat, protože uživateli se vrátí response na nahrání ihned po nahrání a uložení pdf a nemusí s čekat na vygenerování thumbnailu pro pdf.

**Comment by tumovam** [comment]
Automatické generování mi přijde určitě lepší a jistější. To, co popisuješ mi přijde logické. Pokud by fungovalo generování náhledu na pozadí, nemuselo by se čekat na vygenerování hned po přidání PDF a snad by se celý proces zrychlil. Takže pecka! Díky, díky 🥇

**Comment by SykoraLukas** [comment]
Bude se chovat stejně jako pro `eČlánek` tak pro `NDK eČlánek`.

**Comment by tumovam** [comment]
Omlouvám se, že se k tomu musíš pořád vracet, ale ještě mě napadlo... 
U eMonografií se s rychlostí nahrávání zase tak často nesetkáváme (i když jsou PDF větší, tak nepřidávám x PDF "rychle" za sebou). 
Nebylo by ale dobré mít proces na generování thumbnailu sjednocený i u jiných e-modelů, třeba jako je právě model eMonografie? 
Pokud je to moc složité, tak se bez toho obejdeme, jen mi to došlo, když jsi explicitně zmínil eČlánky a NDK eČlánky. 
Díky

**Comment by SykoraLukas** [comment]
Ano, zmínil jsem sice eČlánek a NDK eČlánek, ale týká se to všech NDK eModelů, ke kterým jde přidat PDF. Takže to co chceš je splněné.

**Comment by tumovam** [comment]
Doplnění pro @albertoh pokud nahraji PDF, tak ve Správě procesů je ve slouci Profil "_formField.profiles.externalProfile.thumbnail_" (plus navíc chyba _Missing configuration for import.thumbnail.processor..._)

**Comment by tumovam** [comment]
Lokální test ve v2.4.2/4.3.2 a ve Správě procesů stále tato hláška: 

<img width="580" height="81" alt="Image" src="https://github.com/user-attachments/assets/22f5f3d0-0730-4cda-86ef-e700c46f64b5" />

 Možná lepší třeba **"generování náhledu PDF"** nebo v procesech vůbec nezobrazovat?

**Comment by tumovam** [comment]
Jen doplním, že skončí chybou "java.lang.IllegalStateException: Missing configuration for import.thumbnail.processor"

**Comment by SykoraLukas** [comment]
Konfigurace doplněna. Nyná můžeš zkoušet. otestováno na https://proarc-angular-upgrade.inovatika.dev/repository/uuid:4d590787-88a6-4ea6-9381-53bd2d68fa9a

<img width="2563" height="186" alt="Image" src="https://github.com/user-attachments/assets/bfb32fb5-677c-495d-8394-c3a606f46f50" />

@albertoh prosím doplňte překlad.

**Comment by albertoh** [comment]
Preklad hotov

**Comment by tumovam** [comment]
ve v2.4.2 (30.03.2026) se PDF nahraje, vytvoří se náhled a text ve Správě procesů je taky v pořádku. Děkujeme!
